zombienet_tests/functional/0003-beefy-and-mmr.zndsl

```diff
@@ -38,8 +38,16 @@ validator-0: reports substrate_beefy_best_block is at least 21 within 130 second
 validator-1: reports substrate_beefy_best_block is at least 21 within 130 seconds
 validator-2: reports substrate_beefy_best_block is at least 21 within 130 seconds
 
-# TODO (issue #11972): Custom JS to test BEEFY RPCs
-# TODO (issue #11972): Custom JS to test MMR RPCs
+# Custom JS to test BEEFY RPCs.
```

Comment:
Not really related to this PR, but leaving this here.

I think we could do some more simplifications to this zombienet tests. For example, in the first part:

```
validator-0: reports node_roles is 4
validator-1: reports node_roles is 4
validator-2: reports node_roles is 4
validator-3: reports node_roles is 4
```
I think can be replaced with:
```
validator: reports node_roles is 4
```
which should do the check for the entire group of nodes.

Also we could separate `validator-3` from the `validator` group. So that after we pause `validator-3`, we can still check for example:
```
validator-0: reports substrate_beefy_validator_set_id is at least 1 within 70 seconds
validator-1: reports substrate_beefy_validator_set_id is at least 1 within 70 seconds
validator-2: reports substrate_beefy_validator_set_id is at least 1 within 70 seconds
```
as just 
`validator: reports substrate_beefy_validator_set_id is at least 1 within 70 seconds`

Also this (having all the running nodes in a separate group then validator-3) could help with this PR as well. It would enable us to do `validator: js-script ....` and do the checks only on the `validator` group, without needing to specify the exact nodes.

Please let me know if you're interested in trying this. If not, I can do it. Probably as part of a separate PR anyway.

Reply:
good suggestion - done in https://github.com/paritytech/polkadot/pull/7068/commits/edb0a95d8b92c8cd4fbfefbadf5ef4924c1399d6

Reply:
Thanks ! Looks good ! Now can we also change all checks that look like
```
validator-0: reports node_roles is 4 
validator-1: reports node_roles is 4 
validator-2: reports node_roles is 4
```
to something like 
```
validator: reports node_roles is 4 
```
?

Also did you try to change
`validator: js-script ./0003-beefy-finalized-heads.js with "validator-0,validator-1,validator-2" return is 1 within 5 seconds`
To something like `validator: js-script ./0003-beefy-finalized-heads.js return is 1 within 5 seconds` ? Does it work to get the validators in the js script with something like `networkInfo.getNodes(groupName)` ?

Reply:
thanks, that was a sloppy oversight of mine: https://github.com/paritytech/polkadot/pull/7068/commits/c320c4ad0259e63c0c113662efb36bb6b72722c6

> Also did you try to change [...]? Does it work to get the validators in the js script with something like `networkInfo.getNodes(groupName)`?

Indeed I did, and it does not work since the [`networkInfo`](https://github.com/paritytech/zombienet/blob/6c83458cd043bf047480f6575705706fb6f6d7e0/javascript/packages/orchestrator/src/test-runner/assertions.ts#L198-L223) passed in to scripts [here](https://github.com/paritytech/zombienet/blob/6c83458cd043bf047480f6575705706fb6f6d7e0/javascript/packages/orchestrator/src/test-runner/assertions.ts#L252-L256) is just a projection of [`Network`](https://github.com/paritytech/zombienet/blob/6c83458cd043bf047480f6575705706fb6f6d7e0/javascript/packages/orchestrator/src/network.ts#L77) and, among other aspects, lacks the `getNodes()` function.

If the usage pattern we've introduced here becomes more common, it's a sensible zombienet PR to have `networkInfo` inherit `getNodes()` as well to avoid the unsightly `with "validator-0,validator-1,validator-2"`.